src/error_diffusion.jl

```diff
@@ -96,9 +99,9 @@ function colordither!(
             px = img[r, c]
             alg.clamp_error && (px = clamp01(px))
 
-            col = closest_color(px, cs; metric=metric) # round to closest color
-            out[r, c] = col # apply pixel to dither
-            err = px - col  # diffuse "error" to neighborhood in filter
+            colorindex = argmin(colordiff.(px, labcs; metric=metric)) # find closest color
```

Comment:
I'm not sure if it improves the performance:

```suggestion
            colorindex = argmin(colordiff(px, c; metric=metric) for c in labcs)
```

```julia
julia> X = rand(5) .- 0.5
5-element Vector{Float64}:
  0.03323577517876519
  0.2091426516127396
 -0.09686854335774098
 -0.18712636571366204
 -0.02502486084901634

julia> argmin(abs(x) for x in X)
5

julia> @btime argmin(abs(x) for x in $X)
  14.895 ns (0 allocations: 0 bytes)
5

julia> @btime argmin(abs.($X))
  39.406 ns (1 allocation: 96 bytes)
5

julia> @btime argmin(abs, $X) # Julia-1.7 only
  9.141 ns (0 allocations: 0 bytes)
```